staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/metrics.go

```diff
@@ -58,3 +71,23 @@ func (c *certificateExpirationTracker) set(a *Authenticator, t time.Time) {
 		c.metricSet(&earliest)
 	}
 }
+
+// incrementCallsMetric increments a global metrics counter for the number of calls to an exec
+// plugin, partitioned by exit code. The provided err should be the return value from
+// exec.Cmd.Run().
+func incrementCallsMetric(err error) {
+	switch err.(type) {
+	case nil: // Binary execution succeeded.
+		metrics.ExecPluginCalls.Increment(0, pluginCalledStatus)
+
+	case *exec.ExitError: // Binary execution failed (see "os/exec".Cmd.Run()).
+		metrics.ExecPluginCalls.Increment(err.(*exec.ExitError).ExitCode(), pluginCalledStatus)
+
+	case *exec.Error: // Binary does not exist (see exec.Error).
+		metrics.ExecPluginCalls.Increment(0, pluginNotFoundStatus)
+
+	default: // We don't know about this error type.
+		klog.Warningf("unexpected exec plugin return error type: %T (error = %q)", err, err.Error())
```

Comment:
log lines in a metrics record method are a little unexpected... I'd probably drop this... the only thing it affects is the recorded metric, and people who care will be watching the metric, right?

Reply:
e921dc9049ed37fd53424c2aeaea42df9f4db718